Runtime/Shaders/ComputeShaderHelper.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using UnityEngine;
+
+namespace Unity.MeshSync {
+    internal class ComputeShaderHelper {
+
+        private const string SHADER_CONST_OUTPUT = "Output";
+
+        private readonly ComputeShader shader;
+        int kernelIndex;
+
+        readonly uint groupSizeX;
+        readonly uint groupSizeY;
+
+        Vector2Int maxTextureSize = new Vector2Int(1, 1);
+
+        public ComputeShaderHelper(ComputeShader shader, string kernelName) {
+            this.shader = shader;
+
+            kernelIndex = shader.FindKernel(kernelName);
+            shader.GetKernelThreadGroupSizes(kernelIndex, out groupSizeX, out groupSizeY, out var gsZ);
+        }
+
+        public Texture RenderToTexture(Texture existingTexture) {
+            var renderTarget = existingTexture as RenderTexture;
+
+            // If there is an existing renderTexture, reuse it:
+            if (renderTarget == null ||
+                renderTarget.width != maxTextureSize.x ||
+                renderTarget.height != maxTextureSize.y) {
+                renderTarget = new RenderTexture(maxTextureSize.x, maxTextureSize.y, 32) {
```

Comment:
RenderTextures need to be manually disposed of or they leak memory - see comment about Texture2Ds

There is a RenderTextureDisposeable implementation we used for ScanEngine that could be helpful

https://github.cds.internal.unity3d.com/unity/ScanEngineApp/tree/ab263cbc5104dba8490f1fa6c68809b1ffa52a2c/ScanEngineApp-URP/Assets/Scripts/Common/ImageProcessing

Reply:
I've added code to release it when they're saved as texture2d in `SaveMaterialRenderTexturesToAssetDatabase` and no longer used.
I don't think we can wrap them in a disposable class because they are only set on the material and we don't control its lifecycle except replacing them with textures saved to the asset database. We have no control over the user replacing them with their own texture or removing them from the material.